src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLFunctionsIT.java

```diff
@@ -314,6 +319,66 @@ public void castIntFieldToDoubleWithAliasJdbcFormatGroupByTest() {
         }
     }
 
+    @Test
+    public void castKeywordFieldToDatetimeWithoutAliasJdbcFormatTest() {
+        JSONObject response = executeJdbcRequest("SELECT CAST(date_keyword AS DATETIME) FROM "
+                + TestsConstants.TEST_INDEX_DATE + " ORDER BY date_keyword");
+
+        String date_type_cast = "{\"name\":\"cast_date_keyword\",\"type\":\"date\"}";
+        assertEquals(response.getJSONArray("schema").get(0).toString(), date_type_cast);
+        String[] expectedOutput = new String[] {"Tue Aug 19 07:09:13", "Wed Sep 25 02:04:13"};
+
+        for (int i = 0; i < response.getJSONArray("datarows").length(); ++i) {
+            Assert.assertThat(response.getJSONArray("datarows").getJSONArray(i).getString(0)
+                    .indexOf(expectedOutput[i]), not(-1));
+        }
+    }
+
+    @Test
+    public void castKeywordFieldToDatetimeWithAliasJdbcFormatTest() {
+        JSONObject response = executeJdbcRequest("SELECT CAST(date_keyword AS DATETIME) AS test_alias FROM "
+                + TestsConstants.TEST_INDEX_DATE + " ORDER BY date_keyword");
+
+        String date_type_cast = "{\"name\":\"test_alias\",\"type\":\"date\"}";
+        assertEquals(response.getJSONArray("schema").get(0).toString(), date_type_cast);
+        String[] expectedOutput = new String[] {"Tue Aug 19 07:09:13", "Wed Sep 25 02:04:13"};
+
+        for (int i = 0; i < response.getJSONArray("datarows").length(); ++i) {
+            Assert.assertThat(response.getJSONArray("datarows").getJSONArray(i).getString(0)
+                    .indexOf(expectedOutput[i]), not(-1));
+        }
+    }
+
+    @Test
+    public void castFieldToDatetimeWithWhereClauseJdbcFormatTest() {
+        JSONObject response = executeJdbcRequest("SELECT CAST(date_keyword AS DATETIME) FROM "
+                + TestsConstants.TEST_INDEX_DATE + " WHERE date_keyword IS NOT NULL ORDER BY date_keyword");
```

Comment:
Could you use `CAST` in WHERE, for example `WHERE CAST(date_keyword AS DATETIME) = '2019-12-09 ...'`?

Reply:
Ah yes, I misunderstood your original comment. Will add.

Reply:
This is currently not supported, will have to add this functionality.

Reply:
Okay. Do you know the reason? Because normally functions that we support should be able to work in SELECT and WHERE.

Reply:
Not yet- I'm currently debugging.